src/azure-cli-core/azure/cli/core/decorators.py

```diff
@@ -81,3 +84,25 @@ def _wrapped_func(*args, **kwargs):
                     return fallback_return
         return _wrapped_func
     return _decorator
+
+
+def retry(retry_times=3, interval=0.5, exceptions=Exception):
```

Comment:
This decorator idea is neat. I may consider backport this to MSAL EX. :-)

Regarding the `exceptionS` parameter, do you also intend to have it support accepting a tuple, such as `exceptions=(ValueError, ZeroDivisionError)`? It might just work, but better test it out (at least manually), if you haven't already.

Reply:
Yes, it definitely works.